app/controllers/authenticated/HowYouWillGetYourExpensesController.scala

```diff
@@ -41,10 +46,16 @@ class HowYouWillGetYourExpensesController @Inject()(
   def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
 
-      request.userAnswers.get(TaxYearSelectionPage) match {
-        case Some(taxYearSelection) if taxYearSelection.contains(CurrentYear) && taxYearSelection.length > 1 => Ok(currentAndPreviousYearView("", containsCurrentYearMinus1(taxYearSelection)))
-        case Some(taxYearSelection) if taxYearSelection.contains(CurrentYear) => Ok(currentView(""))
-        case Some(taxYearSelection) => Ok(previousView("", containsCurrentYearMinus1(taxYearSelection)))
+      val redirectUrl = navigator.nextPage(HowYouWillGetYourExpensesPage, NormalMode)(request.userAnswers).url
+      val taxYearSelection: Option[Seq[TaxYearSelection]] = request.userAnswers.get(TaxYearSelectionPage)
+
+      taxYearSelection match {
+        case Some(taxYearSelection) if taxYearSelection.contains(CurrentYear) && taxYearSelection.length > 1 =>
+          Ok(currentAndPreviousYearView(redirectUrl, containsCurrentYearMinus1(taxYearSelection)))
+        case Some(taxYearSelection) if taxYearSelection.contains(CurrentYear) =>
+          Ok(currentView(redirectUrl))
+        case Some(taxYearSelection) =>
+          Ok(previousView(redirectUrl, containsCurrentYearMinus1(taxYearSelection)))
       }
```

Comment:
We don't have  a catch all on the end, should be case _ to SessionExpired or Tech Diffs